src/ui_nbgl.c

```diff
@@ -0,0 +1,433 @@
+  uiContext.abandon_status = "Transaction\nrejected";
+  uiContext.approved_status = "TRANSACTION\nCONFIRMED";
+  uiContext.prompt_cancel_message = "Reject\nTransaction ?";
+  uiContext.prompt = "Sign transaction";
+  ui_start(cb, TRANSACTION_TYPE);
+}
+
+static void ui_message_start(void) {
+  uiContext.abandon_status = "Message\nrejected";
+  uiContext.approved_status = "MESSAGE\nSIGNED";
+  uiContext.prompt_cancel_message = "Reject\nMessage ?";
+  uiContext.prompt = "Sign message";
+  ui_start(NULL, MESSAGE_TYPE);
+}
+
+// Other callbacks
+static void display_pubkey_callback(void) {
+  uiContext.tagValueList.pairs = uiContext.tagValues;
+  uiContext.tagValueList.nbPairs = uiContext.nbPairs;
+
+  uiContext.infoLongPress.icon = &G_coin_config->img_nbgl;
+  uiContext.infoLongPress.longPressText = "Approve";
+  uiContext.infoLongPress.longPressToken = 1;
+  uiContext.infoLongPress.tuneId = TUNE_TAP_CASUAL;
+  uiContext.infoLongPress.text = "Confirm\npublic key";
+
+  nbgl_useCaseStaticReviewLight(&uiContext.tagValueList,
+                                &uiContext.infoLongPress, "Cancel",
+                                status_callback);
+}
+
+// Flow entry point
+static void single_flow_callback(int token, uint8_t index) {
+  UNUSED(index);
+  transaction_review_callback(token);
+}
+
+void ui_confirm_single_flow(void) {
+  uiContext.approved_cb = approved_user_action_processing_callback;
+  uiContext.abandon_cb = abandon_user_action_callback;
+
+  if (!uiContext.transaction_prompt_done) {
+    uiContext.transaction_prompt_done = true;
+
+    ui_transaction_start(ui_confirm_single_flow);
+  } else {
+    snprintf(vars.tmp.feesAmount, sizeof(vars.tmp.feesAmount), "#%d",
+             btchip_context_D.totalOutputs - btchip_context_D.remainingOutputs +
+                 1);
+
+    uiContext.tagValues[0].item = "Output";
+    uiContext.tagValues[0].value = vars.tmp.feesAmount;
+
+    uiContext.tagValues[1].item = "Amount";
+    uiContext.tagValues[1].value = vars.tmp.fullAmount;
+
+    uiContext.tagValues[2].item = "Address";
+    uiContext.tagValues[2].value = vars.tmp.fullAddress;
+
+    uiContext.nbPairs = 3;
+
+    nbgl_pageNavigationInfo_t info = {
+        .activePage =
+            btchip_context_D.totalOutputs - btchip_context_D.remainingOutputs,
+        .nbPages = btchip_context_D.totalOutputs - 1,
+        .navType = NAV_WITH_TAP,
+        .progressIndicator = true,
+        .navWithTap.backButton = false,
+        .navWithTap.nextPageText = "Tap to continue",
+        .navWithTap.nextPageToken = 1,
+        .navWithTap.quitText = "Cancel",
+        .quitToken = 0,
+        .tuneId = TUNE_TAP_CASUAL};
+
+    nbgl_pageContent_t content = {
+        .type = TAG_VALUE_LIST,
+        .tagValueList.nbPairs = uiContext.nbPairs,
+        .tagValueList.pairs = (nbgl_layoutTagValue_t *)uiContext.tagValues};
+    releaseContext();
+    pageContext =
+        nbgl_pageDrawGenericContent(&single_flow_callback, &info, &content);
```

Comment:
Why couldn't you use a static review here ? :thinking:

Reply:
I wanted to be able to specify the page count regarding the number of outputs to be displayed

Reply:
You could probably use [nbgl_useCaseGetNbPagesForTagValueList](https://github.com/LedgerHQ/ledger-secure-sdk/blob/bdb5220a7841d93dcbb83e35a7d128a31d60094f/lib_nbgl/include/nbgl_use_case.h#L92) from the SDK with the static review to achieve your goal.